diff --git a/lib/cartopy/io/__init__.py b/lib/cartopy/io/__init__.py
index 0a6a2c6d..7fee3f33 100644
--- a/lib/cartopy/io/__init__.py
+++ b/lib/cartopy/io/__init__.py
@@ -10,6 +10,17 @@ sub-packages for loading, saving and retrieving various data formats.

 """

+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass
+
 import collections
 from pathlib import Path
 import string
diff --git a/lib/cartopy/io/img_tiles.py b/lib/cartopy/io/img_tiles.py
index a9e92af3..eb13659c 100644
--- a/lib/cartopy/io/img_tiles.py
+++ b/lib/cartopy/io/img_tiles.py
@@ -17,6 +17,17 @@ using tiles in this way can be found at the

 """

+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass
+
 from abc import ABCMeta, abstractmethod
 import concurrent.futures
 import io
diff --git a/lib/cartopy/io/ogc_clients.py b/lib/cartopy/io/ogc_clients.py
index 4e463c0b..9f17bf8f 100644
--- a/lib/cartopy/io/ogc_clients.py
+++ b/lib/cartopy/io/ogc_clients.py
@@ -15,6 +15,17 @@ this way can be found at :ref:`sphx_glr_gallery_web_services_wmts.py`.

 """

+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass
+
 import collections
 import io
 import math
diff --git a/lib/cartopy/io/shapereader.py b/lib/cartopy/io/shapereader.py
index 704b2924..91df19f4 100644
--- a/lib/cartopy/io/shapereader.py
+++ b/lib/cartopy/io/shapereader.py
@@ -27,6 +27,17 @@ geometry representation of shapely:

 """

+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass
+
 import io
 import itertools
 from pathlib import Path
diff --git a/lib/cartopy/io/srtm.py b/lib/cartopy/io/srtm.py
index 1c88d1de..9167e92e 100644
--- a/lib/cartopy/io/srtm.py
+++ b/lib/cartopy/io/srtm.py
@@ -16,6 +16,17 @@ using classes and functions defined below.

 """

+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass
+
 import io
 from pathlib import Path
 import warnings
diff --git a/lib/cartopy/tests/io/test_srtm.py b/lib/cartopy/tests/io/test_srtm.py
index e9d8db2b..fc35c71c 100644
--- a/lib/cartopy/tests/io/test_srtm.py
+++ b/lib/cartopy/tests/io/test_srtm.py
@@ -2,6 +2,17 @@
 #
 # This file is part of Cartopy and is released under the BSD 3-clause license.
 # See LICENSE in the root of the repository for full licensing details.
+# urllib.request is not available in WebAssembly (i.e., Pyodide).
+
+# If pyodide_http is present, we are very likely in WASM environment.
+# pyodide-http provides patches for urllib.
+# See https://github.com/SciTools/cartopy/issues/2514
+try:
+    import pyodide_http
+    pyodide_http.patch_all()
+except:
+    # not in WASM
+    pass

 import numpy as np
 from numpy.testing import assert_array_equal
